### DISCUSSION TITLE
Compiling for Debian Buster Armel (Kirkwood)

### DISCUSSION THREAD

**Discussion by BobManMasher** [pupnp/pupnp#217] [Q&A] [answered]
Hi, New to this and am struggling. I need a new pupnp library so I can make a new Gerbera config. Has anyone either got an outline on how to compile under Debian Buster or a built package for the library? I have downloaded the Tar file and extracted it but can't figure out how to use make to compile it. 
Cheers - Bob

**Comment by Vollstrecker** [comment, marked as answer]
Have you this: https://github.com/pupnp/pupnp#5-build-instructions

**Reply by BobManMasher** [reply]
I had seen it but it made no sense to me.  I have since been able to download the cmake code and run command line on the target system but I now get an error during build 

```
CMake Error at CMakeLists.txt:341 (add_library):
  add_library cannot create ALIAS target "Threads::Shared" because target
  "Threads::Threads" is imported but not globally visible.

CMake Error at CMakeLists.txt:342 (add_library):
  add_library cannot create ALIAS target "Threads::Static" because target
  "Threads::Threads" is imported but not globally visible.

-- Check size of pthread_rwlock_t
-- Check size of pthread_rwlock_t - done
-- Configuring incomplete, errors occurred!
```
Not sure why that should be at the moment - any suggestions?

**Reply by Vollstrecker** [reply]
Nice catch. It seems that the used technic isn't allowed in the version of cmake shipped with debian stable. I need to check if this can be changed (again) to work there without getting to hacky.

For now I can only recommend users of debian-stable or ubuntu 18.04 to use configure.

**Reply by Vollstrecker** [reply]
Fixes for your problem are merged, You want to test it?

**Reply by BobManMasher** [reply]
Thanks - Just cloned the latest version, did a cmake and it completed without error.  Looks like you fixed it - much appreciated.
